### PR TITLE
[DO NOTE MERGE] Move babylon to monorepo

### PR DESCRIPTION
**Pull request by danez**
This is the first attempt to move babylon into the monorepo.
The history has been completely preserved, by merged the two unrelated git histories together.

I haven't done any modifications yet. If anyone wants to help here are the tasks, simply push to the branch or create a PR against this branch. Here are the tasks:

- [x] Convert babylon tests to mocha (describe/it), ensure the run in travis correctly and remove ava
- [x] Figure out how to make test262 work in monorepo
- [x] Figure out how to make flow tests work in monorepo
- [x] Cleanup unnecessary files (travis, github templates, configs, etc)
- [x] Integrate build into lerna publish?
- [x] Align prettier/eslint if needed, maybe codestyle is different, at least ensure all src and test files in babylon get linted/formated, (babylon/scripts should probably excluded or have a different prettier config)
- [x] Cleanup dev dependencies that are not needed anymore
- [x] Update url to babylon (package.json,...)

Other things to do
- [ ] Move Issues and PRs
- [ ] Figure out versioning as babylon has some same versions that babel also has. Either make rc.0 or ceta.0 or increase to next beta (currently 28) which both can share @hzoo 
- [x] Remove all babylon git tags, which are interfering with babel on final merge @danez

The merge in the end needs to be done manually. Will update to the latest babylon changes before merge and replay all the changes we now do in this branch. 

(Github only shows the latest 250 commits, but all are there)

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -20,6 +20,10 @@ experimental/babel-preset-env/data
 experimental/babel-preset-env/test/debug-fixtures
 packages/babel-standalone/babel.js
 packages/babel-standalone/babel.min.js
+packages/babylon/build
+packages/babylon/.nyc_output
+packages/babylon/coverage
```
Comment (danez) [inline .eslintignore]: We will not have coverage reports in the babylon folder, it works exactly the same as all the babel packages.

File: packages/babylon/src/types.js [hunk 2]
```diff
@@ -22,15 +22,15 @@ export type Comment = {
...
-  extra: { [key: string]: any },
+  start: number;
+  end: number;
+  loc: SourceLocation;
+  range: [number, number];
+  leadingComments?: ?Array<Comment>;
+  trailingComments?: ?Array<Comment>;
+  innerComments?: ?Array<Comment>;
+
+  extra: { [key: string]: any };
```
Comment (nicolo-ribaudo) [inline packages/babylon/src/types.js]: I don't know why here there are the semicolons: `make fix` adds them but my editor (which loads the correct eslint config) removes them 🤔

Comment (existentialism) [reply packages/babylon/src/types.js]: Prettier recently forces semi's on interfaces:
https://github.com/prettier/prettier/issues/2593

**Comment by nicolo-ribaudo** [comment]
What should we do with `packages/babylon/LICENSE`? The difference with the root LICENSE file is that in the babylon's the copyright is `by various contributors (see AUTHORS)`, while in the root one it is by @kittens

**Comment by danez** [comment]
I removed `.yarnrc` already. Not sure about the license. We could add both lines to satisfy all parties maybe? Or we leave the license as it is and only update the year. I guess the babylon license is also there because babylon is a fork of acorn.

**Comment by nicolo-ribaudo** [comment]
We should also move `babylon` -> `@babel/babylon`

File: scripts/tests/test262/run_babylon_test262_utils.js [hunk 3]
```diff
@@ -7,10 +7,10 @@ const pfs = {
   readFile: promisify(fs.readFile),
   writeFile: promisify(fs.writeFile),
   readdir: promisify(fs.readdir),
-  stat: promisify(fs.stat)
+  stat: promisify(fs.stat),
 };
 
-const parse = require("..").parse;
+const parse = require("babylon").parse;
```
Comment (existentialism) [inline scripts/tests/test262/run_babylon_test262_utils.js]: Will this resolve parse from packages/babylon?

Comment (danez) [reply scripts/tests/test262/run_babylon_test262_utils.js]: I thought lerna or yarn will symlink, but they don't currently, fixed.

**Comment by nicolo-ribaudo** [comment]
I just pushed some commits; now tests are green again 🎉

**Comment by danez** [comment]
Updated, the only thing not working again is the rollup build for babylon.

**Comment by danez** [comment]
There are three commits:
* This moves babylon into packages/babylon: 2d378d0
* This one merges babylon and babel: 5253741
* This one integrates babylon into babel: c2c787c

Only the last is probably good to review.

**Comment by azz** [comment]
Forgot to remove `[DO NOT MERGE]` from PR title :p

**Comment by xtuc** [comment]
Yes, I hope it won't break the internet

**Pull request merged**